packages/client/src/hooks/useSuccessMessage.js

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useState } from "react";
+import { useModalContext } from "contexts";
+import { SuccessModal } from "modals";
+import { ASSET_TYPES } from "../constants/enums";
+
+export default function useSuccessMessage() {
```

Comment:
In order to avoid confusing this module with a generic success message modal, suggest renaming it to useTransactionSuccessMessage and TransactionSuccessModal,

Reply:
Do we use a generic success modal?

Reply:
If we have a generic error modal we should have one for success

Reply:
Then this is ok, because error modal is triggered when the transaction fails for some reason

Reply:
Then we should name the error modal TransactionErrorModal, using a generic name for a component with specific use case is not scalable, imagine in the future we want to have a generic errorMessage or successMessage, we will have to do a lot of refactoring

Reply:
In case we add a generic success/error message we can rename this one

Reply:
How about moving the logic from this Hook inside of the SuccessModal, and really make this a generic one?

Reply:
I made some modifications - renamed the hook const and the success modal to include transaction in the naming
Also, in the hook I removed useState and useEffect and tested - it looks like it works really well
Can you please take another look?

Reply:
could you help me understand why do we use `hook` instead of a normal react component for this success screen?

Reply:
Discussed with Linxiao on this topic when we implemented the error modal, based on the comment:
`Let’s encapsulate the showError and extractError into a useErrorMessage hook. Utils should not be aware of the components`
So I went on with the same approach here

Reply:
I updated this to work without using hooks.